File: fix_jsmediatags.js
const fs = require('fs');
let code = fs.readFileSync('app/components/MetadataViewer.tsx', 'utf8');

// Remove import * as jsmediatags
code = code.replace("import * as jsmediatags from 'jsmediatags';", "import Script from 'next/script';");

// Use window.jsmediatags
code = code.replace(
  "jsmediatags.read(",
  "const jsmediatags = (window as any).jsmediatags;\n    if (!jsmediatags) { setError('Library not loaded. Please refresh.'); setLoading(false); return; }\n    jsmediatags.read("
);

// Add script tag before Upload Box
code = code.replace(
  "{/* Upload Box */}",
  `<Script src="https://cdnjs.cloudflare.com/ajax/libs/jsmediatags/3.9.5/jsmediatags.min.js" strategy="beforeInteractive" />\n      {/* Upload Box */}`
);

fs.writeFileSync('app/components/MetadataViewer.tsx', code);

File: fix-jsmediatags.py
import re

with open("app/components/MetadataRemover.tsx", "r") as f:
    content = f.read()

# Remove the import line if it exists
content = re.sub(r"import\('jsmediatags'\)\.then\(\(jsmediatags\) => \{.*?(?=\s*\}\)\.catch\()", "", content, flags=re.DOTALL)
# It's better to just regex replace the whole loadMetadata function

old_func = """  const loadMetadata = (fileToRead: File) => {
    // Dynamic import to avoid SSR issues with jsmediatags
    import('jsmediatags').then((jsmediatags) => {
      jsmediatags.default.read(fileToRead, {
        onSuccess: (tag: any) => {
          setMetadata(tag.tags);
        },
        onError: (error: any) => {
          console.error("Error reading tags", error);
          // Set an empty object to indicate we tried and found nothing or failed
          setMetadata({});
        }
      });
    }).catch(err => {
      console.error("Error importing jsmediatags", err);
      setMetadata({});
    });
  };"""

new_func = """  const loadMetadata = (fileToRead: File) => {
    const loadScriptAndRead = () => {
      const jsmediatags = (window as any).jsmediatags;
      if (jsmediatags) {
        jsmediatags.read(fileToRead, {
          onSuccess: (tag: any) => setMetadata(tag.tags),
          onError: (error: any) => {
            console.error("Error reading tags", error);
            setMetadata({});
          }
        });
      } else {
        setMetadata({});
      }
    };

    if (!(window as any).jsmediatags) {
      const script = document.createElement('script');
      script.src = "https://cdnjs.cloudflare.com/ajax/libs/jsmediatags/3.9.5/jsmediatags.min.js";
      script.onload = loadScriptAndRead;
      script.onerror = () => setMetadata({});
      document.body.appendChild(script);
    } else {
      loadScriptAndRead();
    }
  };"""

content = content.replace(old_func, new_func)

with open("app/components/MetadataRemover.tsx", "w") as f:
    f.write(content)
